scripts/memory/weaviate.py

```diff
@@ -0,0 +1,110 @@
+from config import Config
+from memory.base import MemoryProviderSingleton, get_ada_embedding
+import uuid
+import weaviate
+from weaviate import Client
+from weaviate.embedded import EmbeddedOptions
+from weaviate.util import generate_uuid5
+
+def default_schema(weaviate_index):
+    return {
+        "class": weaviate_index,
+        "properties": [
+            {
+                "name": "raw_text",
+                "dataType": ["text"],
+                "description": "original text for the embedding"
+            }
+        ],
+    }
+
+class WeaviateMemory(MemoryProviderSingleton):
+    def __init__(self, cfg):
+        auth_credentials = self._build_auth_credentials(cfg)
```

Comment:
@cs0lar I think adding support for API key would be immensely useful for users getting started using weaviate's free sandbox environment:

https://weaviate.io/developers/weaviate/client-libraries/python#api-key-authentication

Reply:
added